lib/mocha.js

```diff
@@ -234,6 +234,34 @@ Mocha.prototype.loadFiles = function (fn) {
   fn && fn();
 };
 
+/**
+ * Load registered files via es6 dynamic imports.
+ *
+ * Note: `eval` is used in this function or else mocha will not
+ * compile on older node versions.
```

Comment:
I don't understand where this is supposed to work.

Reply:
The new `import()` operator allows ECMAScript Modules to be loaded dynamically. The hope was that tests written as modules (i.e. tests including `import obj from './script.js'` statements) can be loaded into Mocha using dynamic imports without having to first use a transpiler to convert ECMAScript Modules into common js modules.

Reply:
right, but I'm unclear on why we're removing old versions of Node.js from the matrix.

> Current support for modules in node is experimental. It is likely that mocha will want to wait untill node starts properly supporting modules before merging this pr.

if we're going to break backwards compatibility, then it'd have to wait until the oldest version of Node.js that supports modules is no longer maintained.  that'll be a few years.  so, that's why I'm trying to understand what I'm supposed to do with this...

Reply:
Sorry, I didn't mean to add that change to the pull request, I will revert it.

The idea was that mocha could add a new CLI flag which would cause mocha to use the new `import()` instead of `require()` to load test files.

This new CLI flag could only be used on the most recent versions of node but mocha would work normally if this flag was not set.

The tests fail because I added a test for the new flag and travis/appveyor runs this test on old versions of node, I didn't know how to work around that.